core/src/main/scala/org/apache/spark/TaskContextImpl.scala

```diff
@@ -30,6 +30,7 @@ private[spark] class TaskContextImpl(
     override val attemptNumber: Int,
     override val taskMemoryManager: TaskMemoryManager,
     val runningLocally: Boolean = false,
+    val stageAttemptId: Int = 0,  // for testing
```

Comment:
oops, good catch.  this was used in earlier versions of tests before i switched to your test.  thanks, I've updated it now.